### PR TITLE
fix #4003 : added nth fibonacci number code using dynamic programming

### LINKED ISSUE #4003: Add new algorithm for Fibonacci using dynamic programming strategies
**This is a(n):**
- [x] New algorithm
- [ ] Update to an existing algorithm
- [ ] Error
- [ ] Proposal to the Repository

**Details:**
Section     : Dynamic Programming
Topic         : Fibonacci Series using Dynamic Programming
Language : Python 3.x

@arnavb I can work on adding this. Can i go ahead?

### PR DESCRIPTION
**Pull request by yedhink**
**Fixes issue:**
fixes #4003 

**Changes:**
added new algorithm for finding the nth **fibonacci** number using **dynamic programming** using **python3**.

### CODE REVIEW

File: code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from sys import argv
```
Comment (arnavb) [inline code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py]: Could you rename the file to just `fibonacci.py`?

File: code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+# N'th fibonacci number using Dynamic Programming
+def fibonacci(n):
+
```
Comment (arnavb) [inline code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py]: Remove this newline.

File: code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from sys import argv
+
+
+# N'th fibonacci number using Dynamic Programming
```
Comment (arnavb) [inline code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py]: Could you turn this comment into a docstring?

File: code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            arr[n - 1] = fibonacci(n - 1)
+
+        if arr[n - 2] == 0:
+            arr[n - 2] = fibonacci(n - 2)
+
+    arr[n] = arr[n - 2] + arr[n - 1]
+    return arr[n]
+
+
+print(fibonacci(int(argv[1])))
```
Comment (arnavb) [inline code/dynamic_programming/src/fibonacci/fibonacci_dynamic.py]: Could you use an `if __name__ == '__main__'`?

**Comment by yedhink** [comment]
@arnavb have made the necessary changes.

**Comment by yedhink** [comment]
@arnavb Thanks for approving. Okay to be merged?

**Comment by arnavb** [comment]
djldk

**Pull request merged**